lib/Dialect/FIRRTL/Import/FIRParser.cpp

```diff
@@ -5825,10 +5825,24 @@ ParseResult FIRCircuitParser::parseCircuit(
     return ArrayAttr::get(getContext(), layersAttr);
   };
 
+  auto arrayToAttr = [&](const ArrayRef<std::string> strArray) {
+    SmallVector<StringRef> attr;
+    for (const auto &str : strArray)
+      attr.push_back(str);
+    OpBuilder builder(getContext());
+    if (attr.empty())
+      return ArrayAttr();
+    return builder.getStrArrayAttr(attr);
+  };
```

Comment:
This is super close to `parseLayers`. Can these be combined?

Reply:
Merged them.